tools/stellar-sign/main.go

```diff
@@ -1,28 +1,95 @@
 // stellar-sign is a small interactive utility to help you contribute a
-// signature to a transaction envelope.
+// signature to a transaction envelope or verify a transaction.
 //
-// It prompts you for a key
+// It prompts you for a key, public (verify) or private (sign)
 package main
 
 import (
 	"bufio"
+	b64 "encoding/base64"
 	"flag"
 	"fmt"
-	"github.com/stellar/go/keypair"
-	"github.com/stellar/go/network"
 	"io/ioutil"
 	"log"
 	"os"
 	"strings"
 
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/network"
+
 	"github.com/howeyc/gopass"
 	"github.com/stellar/go/txnbuild"
 	"github.com/stellar/go/xdr"
 )
 
+type SignOrVerify struct {
+	verify  bool
+	privKey *keypair.Full
+	pubKey  *keypair.FromAddress
+}
+
+func (router *SignOrVerify) setKey(input string) {
+	var err error
+	if router.verify {
+		var inputPubKey keypair.KP
+		inputPubKey, err = keypair.Parse(input)
+		if err != nil {
+			log.Fatal(err)
+		}
+		router.pubKey = inputPubKey.FromAddress()
+	} else {
+		router.privKey, err = keypair.ParseFull(input)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+func (router *SignOrVerify) do(gentx *txnbuild.GenericTransaction, networkPassphrase string) (string, error) {
+	var newEnv string
+	var err error
+	if router.verify {
+		var signature []byte
+		var body []byte
+		var gentxEnv xdr.TransactionEnvelope
+		gentxEnv, err = gentx.ToXDR()
+		if err != nil {
+			log.Fatal(err)
+		}
+		signature = gentxEnv.Signatures()[0].Signature
+		gentxHash, _ := gentx.Hash(networkPassphrase)
+		body = gentxHash[:]
+		err = router.pubKey.Verify(body, signature)
+	} else {
+		if tx, ok := gentx.Transaction(); ok {
+			tx, err = tx.Sign(networkPassphrase, router.privKey)
+			if err != nil {
+				log.Fatal(err)
+			}
+			newEnv, err = tx.Base64()
+			if err != nil {
+				log.Fatal(err)
+			}
+		} else {
+			tx, _ := gentx.FeeBump()
+			tx, err = tx.Sign(networkPassphrase, router.privKey)
+			if err != nil {
+				log.Fatal(err)
+			}
+			newEnv, err = tx.Base64()
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+	return newEnv, err
+}
+
 var in *bufio.Reader
 
 var infile = flag.String("infile", "", "transaction envelope")
+var verify = flag.Bool("verify", false, "Verify the transaction instead of signing")
+var testnet = flag.Bool("test", false, "Testnet")
```

Comment:
I think `test` can be confusing. Maybe just use `testnet`?
```suggestion
var testnet = flag.Bool("testnet", false, "Verifies or signs tx using testnet passphrase")
```